### PR TITLE
fix(chromium): restore automation launch flag

### LINKED ISSUE #42379: [Bug]: Microsoft SAML login closes page on Edge since 1.60; adding --enable-automation fixes regression
### Version

1.60.0

### Steps to reproduce

# [Bug]: Microsoft SAML login closes page on Edge since 1.60; adding `--enable-automation` fixes regression

## Version

Regression starts with **Playwright 1.60.0**.

* `1.59.1` — ✅ works
* `1.60.0` — ❌ fails
* later stable versions — ❌ fail
* latest `@next` / canary — ❌ fails
* `1.60.0` with `--enable-automation` explicitly added — ✅ works

## Environment

* OS: **Windows 10 win64**
* Node.js: **v22.13.0**
* Browser: **Microsoft Edge 151**
* Browser configuration: `channel: 'msedge'`
* Edge is **organization-managed through enterprise policies**
* The **same installed Edge binary/version** is used for all Playwright versions tested
* Authentication: **Microsoft Entra ID / `login.microsoftonline.com` SAML flow**
* Playwright: Node.js / TypeScript

## Description

After upgrading from Playwright `1.59.1` to `1.60.0`, our Microsoft SAML authentication flow started closing the Playwright page immediately after the final **Sign in** action.

The application, machine and browser environment are unchanged.

The important finding is that explicitly restoring the Chromium/Edge launch argument:

```ts
--enable-automation
```

makes the authentication flow work again with Playwright `1.60.0`.

This appears significant because `--enable-automation` was present in Playwright's Chromium launch arguments in `1.59.1` but is no longer present starting with `1.60.0`.

## Reproduction

Our Playwright configuration normally contains:

```ts
use: {
  channel: 'msedge',
}
```

The test navigates through our application authentication flow to:

```text
https://login.microsoftonline.com/...
```

and performs the Microsoft login.

Simplified example:

```ts
await page.getByLabel('Password').fill(password);

await page
  .getByRole('button', { name: 'Sign in' })
  .click();
```

With Playwright `1.59.1`, Microsoft completes the SAML redirect back to the application normally.

With Playwright `1.60.0` and every newer version tested, the click itself succeeds, but shortly afterwards Playwright reports that the page has been closed.

For diagnostics I added:

```ts
page.on('close', () => {
  console.log('PAGE CLOSED');
});

page.on('crash', () => {
  console.log('PAGE CRASHED');
});

browser?.on('disconnected', () => {
  console.log('BROWSER DISCONNECTED');
});
```

The observed event is:

```text
PAGE CLOSED
```

The failure can subsequently surface as:

```text
Target page, context or browser has been closed
```

depending on which Playwright operation is awaiting the page at that moment.

Originally this surfaced from `waitForResponse()`, but removing all response/token handling did not change the behavior. The page still closes after the Microsoft Sign in action.

## Important finding / workaround

With Playwright `1.60.0`, changing only the launch configuration to:

```ts
use: {
  channel: 'msedge',

  launchOptions: {
    args: ['--enable-automation'],
  },
}
```

fixes the issue.

The Microsoft SAML login and redirect then complete successfully.

Therefore the comparison is:

```text
Same machine
Same managed Edge
Same Edge version
Same application
Same authentication flow

Playwright 1.59.1                         PASS
Playwright 1.60.0                         FAIL
Playwright > 1.60                         FAIL
Playwright @next                          FAIL

Playwright 1.60.0 + --enable-automation   PASS
```

I also tested whether the Edge-specific `msForceBrowserSignIn` feature was responsible by changing the corresponding `--disable-features` configuration.

That did **not** fix the issue.

Only restoring:

```text
--enable-automation
```

has fixed it so far.

## Expected behavior

Microsoft SAML authentication should complete successfully after clicking **Sign in**, and the existing Edge page should continue the redirect back to the application without being closed.

This is the behavior observed with Playwright `1.59.1` and also with Playwright `1.60.0` when `--enable-automation` is explicitly restored.

## Actual behavior

Starting with Playwright `1.60.0`, the page is closed during/after the Microsoft authentication redirect.

Adding `--enable-automation` restores the previous working behavior.

## Additional context

The Microsoft Edge installation is controlled by organization enterprise policies, so changing the browser version is not possible.

However, this makes the regression comparison quite controlled: all Playwright versions are tested against the **same Edge installation and version**.

This seems potentially related to #41348, which reports a Playwright 1.60 regression around Microsoft Entra/OAuth cross-origin navigation and `TargetClosedException`.

The additional observation here is that the regression can be deterministically avoided by restoring `--enable-automation`.

It may be worth investigating whether removing `--enable-automation` in Playwright 1.60 changes Microsoft Edge behavior under enterprise policies during Entra/SAML authentication.

### Additional context

_No response_

### Environment

```shell
System:
  OS: Windows 11 10.0.26100
  CPU: (8) x64 Intel Xeon Processor (Cascadelake)
  Memory: 16.48 GB / 31.88 GB

Binaries:
  Node: 22.13.0 - C:\Program Files\nodejs\node.EXE
  npm: 10.9.2 - C:\Program Files\nodejs\npm.CMD

IDEs:
  VSCode: 1.100.0 - C:\Program Files\Microsoft VS Code\bin\code.CMD

npmPackages:
  @playwright/test: ^1.60.0 => 1.60.0
  playwright-bdd: ^9.2.0 => 9.2.0
  playwright-mail-reporter: ^0.0.10 => 0.0.10
```

### PR DESCRIPTION
**Pull request by dcrousso**
removing `--enable-automation` lets managed Edge hand off the browser process during SAML login

restore the default flag while preserving MCP launch behavior through an internal launch setting

regression from <https://github.com/microsoft/playwright/pull/40190>

fixes <https://github.com/microsoft/playwright/issues/42379>

### CODE REVIEW

File: packages/playwright-core/src/tools/mcp/browserFactory.ts [hunk 1]
```diff
@@ -93,6 +93,7 @@ async function createIsolatedBrowser(config: FullConfig, clientInfo: ClientInfo)
   const browser = await browserType.launch({
     tracesDir,
     ...config.browser.launchOptions,
+    ...{ disableAutomation: true },
```
Comment (yury-s) [inline packages/playwright-core/src/tools/mcp/browserFactory.ts]: You can use `ignoreDefaultArgs` option instead.

File: packages/playwright-core/src/server/chromium/chromiumSwitches.ts [hunk 2]
```diff
@@ -87,6 +87,7 @@ export const chromiumSwitches = (options?: { android?: boolean }) => [
   '--unsafely-disable-devtools-self-xss-warnings',
   // Edge can potentially restart on Windows (msRelaunchNoCompatLayer) which looses its file descriptors (stdout/stderr) and CDP (3/4). Disable until fixed upstream.
   '--edge-skip-compat-layer-relaunch',
+  '--enable-automation',
```
Comment (yury-s) [inline packages/playwright-core/src/server/chromium/chromiumSwitches.ts:90]: It was removed in [this change](https://github.com/microsoft/playwright/pull/40190/changes#diff-b950feb8741e8b9b80de84e34cf408afedaba4acd364b776db42360e6a722901) after some back and forth changes. The idea was that when browser is controlled via remote debugger, the same features are already turned on. `--enable-automation` is a big hammer and we haven't heard any complaints after removing the flag. Let's narrow down what is not working in their scenario and see if we can fix it differently.

**Comment by pavelfeldman** [comment]
This likely has implications we don't want to deal with.

File: packages/playwright-core/src/server/chromium/chromiumSwitches.ts (hunk 2 above)
Comment (dcrousso) [reply packages/playwright-core/src/server/chromium/chromiumSwitches.ts:90]: unless im misreading the code, that change always set `--enable-automation` unless `assistantMode: true`, which was only for MCP, meaning that `--enable-automation` was always on for all other cases

furthermore, it was unconditionally added and has been unconditionally set since b20e87d 7yrs ago

Comment (dcrousso) [reply packages/playwright-core/src/server/chromium/chromiumSwitches.ts:90]: as the original issue demonstrates, Chromium also uses `--enable-automation` to disable process handoff behavior <https://issues.chromium.org/issues/371112535>
